### DISCUSSION TITLE
Should a disabled "shot_group" fire off events such as _hit, _complete, etc?

### DISCUSSION THREAD

**Discussion by bernarma** [missionpinball/missionpinball#148] [Q&A] [answered]
I have multiple shot groups defined (with only one active at any one time) containing one or more shots. My expectation is that when a shot group is disabled (i.e. not active), that it should not be firing off events such as _hit or _complete. Could you confirm that is the expected behavior?

As an example, all my shot groups are defined (with one or more shots defined - one in this example)
```
  sg_section4:
    debug: true
    disable_events: section_change
    enable_events: section4_start
    reset_events: sg_section4_hit_complete
    shots: s1_center_ramp
```

If it is not expected, I am seeing an issue that when I have 4 shot groups defined (all disabled except one), the disabled shot groups are still firing off events causing issues.

I am seeing (which may be contributing to this problem) is when the mode starts the following log entries occur (which I expect this to be in a disabled state from the start):
```
2024-09-03 08:23:15,445 : DEBUG : shot_group.sg_section4 : Shot group is complete with state: None
2024-09-03 08:23:15,445 : INFO : EventManager : Event: ======'sg_section4_complete'====== Args={'state': 'None'}
2024-09-03 08:23:15,446 : INFO : EventManager : Event: ======'sg_section4_None_complete'====== Args={}
```

I can post log files if needed but first wanted to make sure this was undesirable behavior and something that it shouldn't be happening, I'll then get it logged with all the information for troubleshooting.

**Comment by avanwinkle** [comment, marked as answer]
There is some nuance here in that shot groups _themselves_ are not truly disabled or enabled, they are merely a collection of shots and post additional events for those shots. Calling enable on a shot group just enables all of the shots in that group, while disable disables the shots in that group.

If you had a shot present in multiple shot groups, and enabled one of the groups, hitting the shot would indeed trigger hit events for all the shot groups that shot is in. And when the shot group initializes, it will always post its current state so mode startups and lights can correctly configure themselves.

So that's all expected. However if you have a shot that's exclusive to a shot group, and that shot is disabled, and the shot group still posts a **hit** event... that's a bug to investigate.

**Reply by bernarma** [reply]
Thanks for the explanation. I've reworked the configuration and all works as you have explained so indeed not a bug.
